lib/Doctrine/MongoDB/EagerCursor.php

```diff
@@ -237,6 +296,26 @@ public function setReadPreference($readPreference, array $tags = null)
     }
 
     /**
+     * {@inheritDoc}
+     */
+    public function hasNext()
+    {
+        $this->initialize();
+
+        $wasIterating = $this->iterating;
+        $hasNext = $this->getNext() !== null;
+
+        // Reset the internal cursor if we weren't iterating
+        if ($wasIterating) {
+            prev($this->data);
+        } else {
+            $this->iterating = false;
```

Comment:
Is it purposeful that you want to leave the cursor in a "non-iterating" state, while still being initialized?